internal/proxy/rate_limiter.go

```diff
@@ -0,0 +1,75 @@
+package proxy
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/hashicorp/golang-lru/v2"
+	"golang.org/x/time/rate"
+)
+
+type RateLimiter struct {
+	rateLimit  int
+	timeWindow int
```

Comment:
Time window will be better as time.Duration.

Now it unknow units. In computer program natural time units may be seconds or milliseconds.

Plus non-duration type need multiply on unnamed constant 1000 it make to read code more difficult.

Reply:
Done